src/main/java/mate/academy/spring/config/SecurityConfig.java

```diff
@@ -1,39 +1,45 @@
 package mate.academy.spring.config;
 
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.context.annotation.Bean;
+import org.springframework.http.HttpMethod;
 import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
 import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
-import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.core.userdetails.UserDetailsService;
 import org.springframework.security.crypto.password.PasswordEncoder;
 
 @EnableWebSecurity
 public class SecurityConfig extends WebSecurityConfigurerAdapter {
+    public static final String ADMIN_ROLE = "ADMIN";
+    public static final String USER_ROLE = "USER";
+    private final UserDetailsService userDetailsService;
+    private final PasswordEncoder passwordEncoder;
+
+    public SecurityConfig(UserDetailsService userDetailsService, PasswordEncoder passwordEncoder) {
+        this.userDetailsService = userDetailsService;
+        this.passwordEncoder = passwordEncoder;
+    }
+
     @Autowired
     public void configure(AuthenticationManagerBuilder auth) throws Exception {
-        auth
-                .inMemoryAuthentication()
-                .passwordEncoder(getEncoder())
-                .withUser("1@123").password(getEncoder().encode("123")).roles("USER");
+        auth.userDetailsService(userDetailsService).passwordEncoder(passwordEncoder);
     }
 
     protected void configure(HttpSecurity http) throws Exception {
-        http
-                .authorizeRequests()
-                .anyRequest().authenticated()
-                .and()
-                .formLogin()
-                .permitAll()
-                .and()
-                .httpBasic()
-                .and()
-                .csrf().disable();
-    }
-
-    @Bean
-    public PasswordEncoder getEncoder() {
-        return new BCryptPasswordEncoder();
+        http.authorizeRequests()
+            .antMatchers(HttpMethod.POST, "/orders/**").hasAnyRole(ADMIN_ROLE, USER_ROLE)
+            .antMatchers("/users/**").hasRole(ADMIN_ROLE)
+            .antMatchers("/shopping-carts/**").hasAnyRole(ADMIN_ROLE, USER_ROLE)
+            .antMatchers(HttpMethod.GET).hasAnyRole(ADMIN_ROLE, USER_ROLE)
```

Comment:
Does it mean that users with role USER can get info about ```/users```
You also add permission, not for all endpoints, I can’t see /movies, /movie-sessions, etc. Let’s  check all endpoints

Reply:
No, user can't get info about /users. Only admin.
I added permissions to all endpoints with get-methods, so I not needed list all endpoint here, only specific